src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -29,16 +29,16 @@ ClusterTaskManager::ClusterTaskManager(
     std::shared_ptr<ClusterResourceScheduler> cluster_resource_scheduler,
     internal::NodeInfoGetter get_node_info,
     std::function<void(const RayTask &)> announce_infeasible_task,
-    std::shared_ptr<ILocalTaskManager> local_task_manager,
+    ILocalTaskManager &local_task_manager,
```

Comment:
Actually this means the destructor order now matters: we need to make sure local_task_manager is destructed after cluster_task_manager

Reply:
Not a problem.

Here's the construction order:
https://github.com/ray-project/ray/blob/2688cdefe8d75dc8735eb95bf03b1140af314716/src/ray/raylet/scheduling/cluster_task_manager.cc#L38-L41
which means other two data members relies on `local_task_manager`.

Destruction order is the reverse order of construction;
And no other data member relies on task manager.